File: NANDOR/Model/URLSession.m
//
//  URLSession.m
//  NANDOR
//
//  Created by Chris Nandor on 2014.05.31.
//  Copyright (c) 2014 Pudge.Net. All rights reserved.
//

#import "URLSession.h"

@interface URLSession ()

@property (strong, nonatomic) NSURLSessionConfiguration *configuration;
@property (strong, nonatomic) NSURLSession *session;
@property (nonatomic) BOOL inited;

@end


@implementation URLSession

- (instancetype)init {
    [self initSession];
    return self;
}

- (void)initSession {
    if (self.inited) {
        return;
    }
    self.configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
    self.session = [NSURLSession sessionWithConfiguration:self.configuration];
    self.inited = true;
    return;
}

- (void)makeRequest:(NSString *)baseURL with:(NSString *)params {
    //NSLog(@"URL: %@?%@", baseURL, params);
    //return;

    NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"%@?%@", baseURL, params]];
    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
    
    NSURLSessionDataTask *dataTask = [self.session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
    }];
    [dataTask resume];
}

@end

File: NANDOR/Model/URLSession.swift
//
//  URLSession.swift
//  NANDOR
//
//  Created by Chris Nandor on 2014.06.02.
//  Copyright (c) 2014 Pudge.Net. All rights reserved.
//

import Foundation

var configuration = NSURLSessionConfiguration();
var session = NSURLSession(configuration: configuration)

class URLSession:NSObject {
    func makeRequest(baseURL:String, with:String) {
        NSLog("%@?%@", baseURL, with)
        return

        var url = NSURL(string: "\(baseURL)?\(with)")
        var dataTask = session.dataTaskWithHTTPGetRequest(url)
        dataTask.resume()
    }
}